eclipsesource/theia
# preferences: fix merge method in PreferenceProvider

#### What it does

When merging preferences, in the PreferenceProvider, also merge properties of type array. Currently, properties of type object are merged, but for arrays only the value from the target JSON is used.

This fixes the first use case described in bug 8987, where launch configs defined in the workspace are not visible, if other launch configs exist in a `launch.json` file.

#### How to test

1. Create a workspace with a test folder and a `.theia-workspace` file, e.g.:
```
{
    "folders": [
        {
            "path": "test"
        }
    ],
    "settings": {
        "launch": {
            "configurations": [
                {
                    "name": "from .theia-workspace",
                    "request": "launch",
                    "type": "node",
                }
            ]
        }
    }
}
```
2. Go to the `Debug` view and check that the launch configs from the workspace file are present in the dropdown at the top of the view. 
3. Add a `.theia/launch.json` file with at least one launch configuration, e.g.: 
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "from .theia",
            "request": "launch",
            "type": "node",
        }
    ]
}
```
4. Go to the `Debug` view and check that the launch configs from  `launch.json` are visible in the dropdown, alongside the ones from the workspace file.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
